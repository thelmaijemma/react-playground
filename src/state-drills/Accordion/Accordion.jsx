import React, { Component } from 'react';
// see solution version at bottom 

class Accordion extends Component {
  // you only use the constructor when are using props
  constructor(props) {
    super(props)
  }
  state = {
    content: null,
  }

  showGoods = (index) => {
    const {sections} = this.props;
    this.setState({ content: sections[index].content })
  }
  {/*showGoods = (index) => {
    this.setState({ content: this.props.sections[index].content })
  }*/}

  willShowQ = () => {
    const currentContent = this.state.content;

    if (this.state.content) {
      return (<p>
        {currentContent}
      </p>
      )
    }
  }


  renderLi(){
    return this.props.sections.map((section, index) => (
      <li key={index}>
        <button  onClick={() => this.showGoods(index)}>
          {section.title}
        </button>
      </li>
    ))
  }

  render() {
    return (
      <div>
        <ul>
          {this.renderLi()}
        </ul>
        {this.willShowQ()}
      </div>
    )
  }
}

export default Accordion;

/* Solution
import React, { Component } from 'react'
import './Accordion.css'

export default class Accordion extends Component {
  static defaultProps = {
    sections: []
  };

  state = {
    activeSectionIndex: null,
  }

  handleSetActiveSecion = (sectionIndex) => {
    this.setState({ activeSectionIndex: sectionIndex })
  }

  renderItem(section, idx, activeSectionIndex) {
    return (
      <li className='Accordion__item' key={idx}>
        <button
          type='button'
          onClick={() => this.handleSetActiveSecion(idx)}
        >
          {section.title}
        </button>
        {(activeSectionIndex === idx) && <p>{section.content}</p>}
      </li>
    )
  }

  render() {
    const { activeSectionIndex } = this.state
    const { sections } = this.props
    return (
      <ul className='Accordion'>
        {sections.map((section, idx) =>
          this.renderItem(section, idx, activeSectionIndex)
        )}
      </ul>
    )
  }
}
*/