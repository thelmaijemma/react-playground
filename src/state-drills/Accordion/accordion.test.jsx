  // ACCORDION TESTING
  import React from 'react'
  import { shallow } from 'enzyme'
  import toJson from 'enzyme-to-json'
  import Accordion from './Accordion';
  
  const sections = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ]
  describe(`Accordion Component`, () => {
   
  
    it('The component opens a clicked section', () => {
      const wrapper = shallow(<Accordion sections={sections} />)
      wrapper.find('button').at(1).simulate('click')
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  
    it('The component renders an empty li when the sections prop is not supplied.', () => {
      const wrapper = shallow(<Accordion />)
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  
    it('The component renders no sections as active by default', () => {
      const wrapper = shallow(<Accordion sections={sections} />)
      expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('The component opens only a single clicked section', () => {
        const wrapper = shallow(<Accordion sections={sections} />)
        wrapper.find('button').at(1).simulate('click')
        wrapper.find('button').at(2).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
      })
  })