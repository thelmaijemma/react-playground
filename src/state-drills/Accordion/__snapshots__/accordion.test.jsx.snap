// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Component The component opens a clicked section 1`] = `
<div>
  <ul>
    <li>
      <button
        key="0"
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li>
      <button
        key="1"
        onClick={[Function]}
      >
        Section 2
      </button>
    </li>
    <li>
      <button
        key="2"
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
  <p>
    Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
  </p>
</div>
`;

exports[`Accordion Component The component opens only a single clicked section 1`] = `
<div>
  <ul>
    <li>
      <button
        key="0"
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li>
      <button
        key="1"
        onClick={[Function]}
      >
        Section 2
      </button>
    </li>
    <li>
      <button
        key="2"
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
  <p>
    Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
  </p>
</div>
`;

exports[`Accordion Component The component renders no sections as active by default 1`] = `
<div>
  <ul>
    <li>
      <button
        key="0"
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li>
      <button
        key="1"
        onClick={[Function]}
      >
        Section 2
      </button>
    </li>
    <li>
      <button
        key="2"
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;

exports[`Tabs Component renders empty given no tabs 1`] = `<div />`;

exports[`Tabs Component renders the first tab by default 1`] = `<div />`;

exports[`closes the first tab and opens any clicked tab 1`] = `
<div>
  <button
    key="0"
    onClick={[Function]}
  >
    First tab
  </button>
  <button
    key="1"
    onClick={[Function]}
  >
    Second tab
  </button>
  <button
    key="2"
    onClick={[Function]}
  >
    Third tab
  </button>
  <div
    className="content"
  >
    Laboriosam exercitationem quos consectetur expedita consequatur. Fugit, sapiente aspernatur corporis velit, dolor eum reprehenderit provident ipsam, maiores incidunt repellat! Facilis, neque doloremque. Lorem ipsum dolor sit amet consectetur adipisicing elit.
  </div>
</div>
`;
